projects/optic/src/utils/write-optic-config.ts

```diff
@@ -0,0 +1,21 @@
+import yaml from 'yaml';
+import fs from 'fs';
+import { logger } from '../logger';
+
+// merges a yaml document into the existing optic.yml and writes it to disk. if newDocument's top-level
+// key already exists in the Optic config, it is overwritten.
+export function updateOpticConfig(
+  newDocument: yaml.Document.Parsed,
+  opticConfigPath: string
+) {
+  try {
+    let opticYml = yaml.parseDocument(fs.readFileSync(opticConfigPath, 'utf8'));
```

Comment:
Instead of `readFileSync`, we should use `await fs.readFile` which is exported from `import fs from 'node:fs/promises'`